contracts/multiswap-contracts/FiberRouter.sol

```diff
@@ -228,116 +237,71 @@ contract FiberRouter is Ownable, TokenReceivable {
+        require(sd.targetAddress != address(0), "FR: Target address cannot be zero");
         require(amount != 0, "FR: Amount must be greater than zero");
         require(withdrawalData != 0, "FR: Withdraw data cannot be empty");
         require(msg.value != 0, "FR: Gas Amount must be greater than zero");
 
-        // Transfer the gas fee to the gasWallet
-        (bool success, ) = payable(gasWallet).call{value: msg.value}("");
-        require(success, "FR: Gas fee transfer failed");
+        // Transfer tokens to FiberRouter
+        amount = SafeAmount.safeTransferFrom(token, _msgSender(), address(this), amount);
+
+        // Distribute the fees
+        uint256 amountOut = _distributeFees(token, amount, fd);
 
         // Perform the token swap based on swapCCTP flag
+        uint64 depositNonce;
         if (cctpType) {
-            TargetNetwork storage target = targetNetworks[targetNetwork];
-            require(target.targetNetworkDomain != 0, "FR: Target network not found");
-            require(target.targetCCTPFundManager != address(0), "FR: Target CCTP FundManager address not found");
-            require(token == usdcToken, "FR: Only USDC deposits allowed for CCTP swaps");
             // Proceed with the CCTP swap logic
-            amount = SafeAmount.safeTransferFrom(token, _msgSender(), address(this), amount);
-            uint64 depositNonce = _swapCCTP(amount, token, target.targetNetworkDomain, target.targetCCTPFundManager);
-
-            emit CCTPSwap(
-                token,
-                amount,
-                block.chainid,
-                target.targetNetworkDomain,
-                _msgSender(),
-                target.targetCCTPFundManager,
-                depositNonce
-            );
+            SafeERC20.safeTransfer(IERC20(token), cctpFundManager, amountOut);
+            depositNonce = CCTPFundManager(cctpFundManager).swapCCTP(amountOut, token, sd.targetNetwork);
         } else {
             // Proceed with the normal swap logic
-            amount = SafeAmount.safeTransferFrom(token, _msgSender(), pool, amount);
-            amount = FundManager(pool).swapToAddress(
-                token,
-                amount,
-                targetNetwork,
-                targetAddress
-            );
-
-            // Emit normal swap event
-            emit Swap(
+            SafeERC20.safeTransfer(IERC20(token), fundManager, amountOut);
+            FundManager(fundManager).swapToAddress(
                 token,
-                targetToken,
-                block.chainid,
-                targetNetwork,
                 amount,
-                _msgSender(),
-                targetAddress,
-                amount,
-                withdrawalData,
-                msg.value
+                sd.targetNetwork,
+                sd.targetAddress
             );
         }
+
+        // Transfer the gas fee to the gasWallet
+        (bool success, ) = payable(gasWallet).call{value: msg.value}("");
```

Comment:
Use the SafeAmount library for consistency

Reply:
Sure, replaced the previous gas fee transfer code with safeAmount library: 

      `  SafeAmount.safeTransferETH(gasWallet, msg.value);`